drivers/misc/nxp_s32_emios/nxp_s32_emios.c

```diff
@@ -87,11 +91,42 @@ static int nxp_s32_emios_init(const struct device *dev)
 		.masterBusConfig = &nxp_s32_emios_##n##_master_bus_config			\
 	};
 
+#define EMIOS_PWM_INTERRUPT_NAME(name)	DT_CAT3(EMIOS, name, _IRQ)
+
+/*
+ * The real interrupt handlers only defined in some circumstances, just add
+ * weak implementations to avoid populating so many preprocessor directives
+ */
+#define EMIOS_PWM_INTERRUPT_DEFINE(node_id, prop, idx)						\
+	__weak void EMIOS_PWM_INTERRUPT_NAME(DT_STRING_TOKEN_BY_IDX(node_id, prop, idx))(void) {}
+
+#define NXP_S32_EMIOS_PWM_INTERRUPT_DEFINE(n)							\
+	DT_INST_FOREACH_PROP_ELEM(n, interrupt_names, EMIOS_PWM_INTERRUPT_DEFINE)
+
+#define EMIOS_PWM_INTERRUPT_CONFIG(node_id, prop, idx)						\
+	do {											\
+		IRQ_CONNECT(DT_IRQ_BY_IDX(node_id, idx, irq),					\
+			    DT_IRQ_BY_IDX(node_id, idx, priority),				\
+			    EMIOS_PWM_INTERRUPT_NAME(DT_STRING_TOKEN_BY_IDX(node_id, prop, idx)),\
+			    DEVICE_DT_GET(node_id),						\
```

Comment:
Where are these interrupt handlers implemented?

Reply:
Those are defined in HAL: https://github.com/zephyrproject-rtos/hal_nxp/blob/master/s32/drivers/s32k3/Mcl/src/Emios_Mcl_Ip_Irq.c#L542